deployment/src/main/java/io/quarkus/jgit/deployment/GiteaDevServiceInfoBuildItem.java

```diff
@@ -1,38 +1,62 @@
 package io.quarkus.jgit.deployment;
 
+import java.net.URI;
+import java.net.URL;
+
 import io.quarkus.builder.item.SimpleBuildItem;
 
 /**
  * A build item that represents the information required to connect to a Gitea dev service.
  */
 public final class GiteaDevServiceInfoBuildItem extends SimpleBuildItem {
 
+    private final URL url;
     private final String host;
     private final int httpPort;
     private final String adminUsername;
     private final String adminPassword;
 
-    public GiteaDevServiceInfoBuildItem(String host, int httpPort, String adminUsername, String adminPassword) {
+    public GiteaDevServiceInfoBuildItem(String url, String adminUsername, String adminPassword) {
+        this(createURL(url), adminUsername, adminPassword);
+    }
+
+    public GiteaDevServiceInfoBuildItem(URL url, String adminUsername, String adminPassword) {
+        this(url, url.getHost(), url.getPort(), adminUsername, adminPassword);
+    }
+
+    public GiteaDevServiceInfoBuildItem(URL url, String host, int httpPort, String adminUsername, String adminPassword) {
+        this.url = url;
         this.host = host;
         this.httpPort = httpPort;
         this.adminUsername = adminUsername;
         this.adminPassword = adminPassword;
     }
 
-    public int httpPort() {
-        return httpPort;
+    private static URL createURL(String url) {
+        try {
+            return new URI(url).toURL();
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid URL: " + url, e);
+        }
     }
 
-    public String host() {
```

Comment:
I purposedly haven't used the getter syntax because this can be a record in the future

Reply:
Ok, will align to the record style